core/src/wallet/cosmos/api_impl/CosmosLikeTransactionApi.cpp

```diff
@@ -340,5 +340,99 @@ namespace ledger {
             return *this;
         }
 
+        std::string CosmosLikeTransactionApi::serializeTransfer() const {
+            auto document = rapidjson::Document();
+            document.SetObject();
+            auto& allocator = document.GetAllocator();
+
+            auto baseReq = rapidjson::Value(rapidjson::kObjectType);
+            for (const auto &message : _txData.messages) {
+                const auto tmpMessage = std::make_shared<CosmosLikeMessage>(message);
+                if (!(tmpMessage->getMessageType() == api::CosmosLikeMsgType::MSGSEND)) continue;
+
+                auto buffer = rapidjson::Value(rapidjson::kStringType);
+                auto makeStringValue = [&allocator](const std::string &str) {
+                    auto res = rapidjson::Value(rapidjson::kStringType);
+                    res.SetString(str.c_str(), static_cast<rapidjson::SizeType>(str.size()), allocator);
+                    return res;
+                };
+
+                const auto msgSend = CosmosLikeMessage::unwrapMsgSend(tmpMessage);
+
+                buffer = makeStringValue(msgSend.fromAddress);
+                baseReq.AddMember(cosmos::constants::kFrom, buffer, allocator);
+
+                buffer = makeStringValue(_txData.memo);
+                baseReq.AddMember(cosmos::constants::kMemo, buffer, allocator);
+
+                buffer = makeStringValue(networks::getCosmosLikeNetworkParameters(_currency.name).ChainId);
+                baseReq.AddMember(cosmos::constants::kChainId, buffer, allocator);
+
+                buffer = makeStringValue(_accountNumber);
+                baseReq.AddMember(cosmos::constants::kAccountNumber, buffer, allocator);
+
+                buffer = makeStringValue(_accountSequence);
+                baseReq.AddMember(cosmos::constants::kSequence, buffer, allocator);
+
+                buffer = makeStringValue(_txData.fee.gas.toString());
+                baseReq.AddMember(cosmos::constants::kGas, buffer, allocator);
+
+                // todo : remove duplicated code
+                auto getAmountObject = [&] (const std::string &denom, const std::string &amount) {
+                    auto tmp = rapidjson::Value(rapidjson::kStringType);
+                    auto amountObject = rapidjson::Value(rapidjson::kObjectType);
+                    tmp.SetString(amount.c_str(), static_cast<rapidjson::SizeType>(amount.length()), allocator);
+                    amountObject.AddMember(cosmos::constants::kAmount, tmp, allocator);
+                    tmp.SetString(denom.c_str(), static_cast<rapidjson::SizeType>(denom.length()), allocator);
+                    amountObject.AddMember(cosmos::constants::kDenom, tmp, allocator);
+                    return amountObject;
+                };
```

Comment:
Looks like the only capture is the allocator. We might be able to extract that into a real function later

Reply:
I have done that in the explorer, this code will be removed, I just forgot to do it